### PR TITLE
Fix UI issue 5777 Root disk size is not shown as 'Disk Size' on VM deployment.

### PR DESCRIPTION
**Pull request by GabrielBrascher**
### Description

In the UI, when selecting a service offering that has a root disk enforced, stills show the root disk size of the template (or the custom size gave by the user). This PR fixes this issue, that thas been reported in #5777.
Fixes: #5777

Now the UI dynamically changes the root disk size according to the following variables, following the priority order:
1. Service offering size (highest priority)
2. Custom size set by the user
3. Template size

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

#### 1. Choose template, set custom rootdisk size to 123GB
image

#### 2. Choose an offering with root disk size set to 30
image

#### 3. Change to another offering, e.g. with 50G
image

### How Has This Been Tested?

1. Build UI (npm run serve)
2. Choose a template -- root disk size matches with the template
3. Set custom root disk size to 123GB -- UI updated with the new root disk size
4. Choose an offering with root disk size set to 30 -- UI changes to the service offering root disk size
5. Change to another offering, e.g. with 50G -- again, the UI updates

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@GabrielBrascher this works but in my test once I select compute offering with root disk size the Override Root Disk Size option becomes disabled and it remains disabled even after I switch to different offering. Though this isn't particularly related to the PR

File: ui/src/views/compute/DeployVM.vue [hunk 1]
```diff
@@ -1133,6 +1133,10 @@ export default {
         this.vm.disksizetotalgb = this.diskSize
       }
 
+      if (this.serviceOffering.rootdisksize) {
+        this.vm.disksizetotalgb = this.serviceOffering.rootdisksize
+      }
```
Comment (sureshanaparti) [inline ui/src/views/compute/DeployVM.vue:1138]: @GabrielBrascher just a suggestion

```
       if (this.serviceOffering.rootdisksize) {
         this.vm.disksizetotalgb = this.serviceOffering.rootdisksize
       } else if (this.diskSize) {
         this.vm.disksizetotalgb = this.diskSize
       }
```

Comment (GabrielBrascher) [reply ui/src/views/compute/DeployVM.vue:1138]: Your suggestion looks better indeed, I will add it into the PR.
Thanks for the review, @sureshanaparti.

**Comment by sureshanaparti** [comment]
@blueorangutan ui

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Review by utchoang — APPROVED** [review]
LGTM!

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**